core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala

```diff
@@ -82,6 +82,13 @@ class ListOffsetsIntegrationTest extends KafkaServerTestHarness {
   def testThreeRecordsInSeparateBatch(quorum: String): Unit = {
     produceMessagesInSeparateBatch()
     verifyListOffsets()
+
+    // test LogAppendTime case
+    val props: Properties = new Properties()
+    props.setProperty(TopicConfig.MESSAGE_TIMESTAMP_TYPE_CONFIG, "LogAppendTime")
+    createTopicWithConfig(topicNameWithCustomConfigs, props)
+    produceMessagesInSeparateBatch(topic = topicNameWithCustomConfigs)
+    verifyListOffsets(topic = topicNameWithCustomConfigs, 2)
```

Comment:
It's fine to add this test, but it doesn't require the fix in LogValidator in this PR, right? If each batch has only a single record, the offset for maxTimestamp is always the same between logAppend and create time. 

It seems that we need to add a test case that includes multiple records in a non-compressed batch.

Reply:
thanks for the comment!
yes, we should have a test for non-compressed type, where multiple records are sent in one batch, then test the offset to verify the fix for LogValidator. Let me updated this.

Reply:
it would be nice to add comments for this case.

Reply:
agree, let me add that. Thanks for the suggestion!